lib/rubocop/cop/rspec/nested_groups.rb

```diff
@@ -97,30 +97,34 @@ class NestedGroups < Cop
           "Configuration key `#{DEPRECATED_MAX_KEY}` for #{cop_name} is " \
           'deprecated in favor of `Max`. Please use that instead.'
 
-        def_node_search :find_contexts, ExampleGroups::ALL.block_pattern
+        def_node_search :find_example_groups, ExampleGroups::ALL.block_pattern
+        def_node_matcher :example_group?, ExampleGroups::ALL.block_pattern
 
         def on_top_level_describe(node, _args)
-          find_nested_contexts(node.parent) do |context, nesting|
+          find_nested_example_groups(node.parent) do |example_group, nesting|
             self.max = nesting
             add_offense(
-              context.send_node,
+              example_group.send_node,
               message: message(nesting)
             )
           end
         end
 
         private
 
-        def find_nested_contexts(node, nesting: 1, &block)
-          find_contexts(node) do |nested_context|
-            yield(nested_context, nesting) if nesting > max_nesting
+        def find_nested_example_groups(node)
+          find_example_groups(node) do |example_group|
+            nesting = nesting_count(example_group)
 
-            nested_context.each_child_node do |child|
-              find_nested_contexts(child, nesting: nesting + 1, &block)
-            end
+            yield(example_group, nesting) if nesting > max_nesting
           end
         end
 
+        def nesting_count(node)
+          count = node.each_ancestor(:block).count { |n| example_group?(n) }
+          count + 1
```

Comment:
It's really only a matter of personal preference, but maybe:
```ruby
node.each_ancestor(:block).count { |n| example_group?(n) } + 1
```
it's pretty explanatory what it does.

I also see Maxim proposing `count(&method(:example_group?))`, or insisting on a more explanatory variable name for `n`, e.g. `block` (which reflects what it is, but provokes confusion with `&block`).